Makefile

```diff
@@ -180,7 +182,9 @@ generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
 
 check-generate: generate
-	@test -z $(shell git status --untracked-files=no --porcelain)
+	@if [ $(shell git status --untracked-files=no --porcelain | wc -l ) != 0 ]; then \
+  		echo "Error: generated files are out of sync, please run 'make generate' before committing" ; \
```

Comment:
Oh wait - I know we just merged this but I didn't catch that this won't actually count as an error for `make`. We need to return a non-zero error code, not _just_ print an error message. Sorry for missing that. 

Maybe use something like https://www.gnu.org/software/make/manual/html_node/Make-Control-Functions.html?

Reply:
You can probably fix this in the same PR you're using to test to see whether it works or not. AIUI, the presubmit uses the version of the makefile and ci-presubmit.sh (or whatever it's called) that's in the PR, not the one that was previously checked in.